troposphere/static/js/components/images/list/ImageListView.jsx

```diff
@@ -36,7 +36,7 @@ export default React.createClass({
         if (newProps.query != this.props.query) {
             this.setState({
                 query: newProps.query || ""
-            });
+            }, this.updateState);
```

Comment:
The only fetchWhere that includes the user's query is located in updateState. updateState must be called when we get a new query. We get a query 2 ways, `onSearchChange`, or in this method `componentWillReceiveProps`, thus we must also call `updateState` here.